CourseGrab/Home/HomeTableViewCell.swift

```diff
@@ -8,7 +8,152 @@
+    private let enrollButton = UIButton(type: .roundedRect)
+    private let removeButton = UIButton(type: .roundedRect)
+    private let sectionLabel = UILabel()
+    private let statusBadge = UIImageView()
+    private let titleLabel = UILabel()
+    
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        
+        // setup appearance
+        
+        selectionStyle = .none
+        
+        container.translatesAutoresizingMaskIntoConstraints = false
+        container.clipsToBounds = false
+        container.layer.cornerRadius = 5
+        container.layer.shadowColor = UIColor.black.cgColor
+        container.layer.shadowOpacity = 0.3
+        container.layer.shadowRadius = 2
+        container.layer.shadowOffset = .zero
+        container.backgroundColor = .white
+        contentView.addSubview(container)
+        
+        courseLabel.translatesAutoresizingMaskIntoConstraints = false
+        courseLabel.font = .boldSystemFont(ofSize: 14)
+        courseLabel.textColor = .courseGrabGray
+        container.addSubview(courseLabel)
+        
+        enrollButton.translatesAutoresizingMaskIntoConstraints = false
+        enrollButton.layer.cornerRadius = 2
+        enrollButton.setTitleColor(.white, for: .normal)
+        enrollButton.setTitle("ENROLL", for: .normal)
+        enrollButton.titleLabel?.font = .systemFont(ofSize: 12)
+        enrollButton.backgroundColor = .black
+        container.addSubview(enrollButton)
+        
+        removeButton.translatesAutoresizingMaskIntoConstraints = false
+        removeButton.layer.cornerRadius = 2
+        removeButton.setTitleColor(.courseGrabRuby, for: .normal)
+        removeButton.setTitle("REMOVE", for: .normal)
+        removeButton.titleLabel?.font = .systemFont(ofSize: 12)
+        removeButton.layer.borderColor = UIColor.courseGrabRuby.cgColor
+        removeButton.layer.borderWidth = 1
+        container.addSubview(removeButton)
+        
+        sectionLabel.translatesAutoresizingMaskIntoConstraints = false
+        sectionLabel.font = .boldSystemFont(ofSize: 14)
+        sectionLabel.textColor = .courseGrabGray
+        container.addSubview(sectionLabel)
+        
+        statusBadge.translatesAutoresizingMaskIntoConstraints = false
+        statusBadge.contentMode = .scaleAspectFit
+        container.addSubview(statusBadge)
+        
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.font = .systemFont(ofSize: 16)
+        titleLabel.numberOfLines = 0
+        container.addSubview(titleLabel)
+        
+        // setup constraints
+
+        container.snp.makeConstraints { make in
+            make.topMargin.equalTo(contentView).offset(6)
+            make.bottomMargin.equalTo(contentView).inset(6)
+            make.leadingMargin.equalTo(20)
+            make.trailingMargin.equalTo(contentView).inset(20)
```

Comment:
use `equalToSuperview` instead of `contentView`
also why are you using `topMargin` instead of `top`?

Reply:
Also why do we use leadingMargin instead of just leading

Reply:
bump on this ^